src/Tests/AzureClientMagicTests.cs

```diff
@@ -0,0 +1,189 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Jupyter.Core;
+using Microsoft.Quantum.IQSharp;
+using Microsoft.Quantum.IQSharp.AzureClient;
+
+namespace Tests.IQSharp
+{
+    public static class AzureClientMagicTestExtensions
+    {
+        public static void Test(this MagicSymbol magic, string input, ExecuteStatus expected = ExecuteStatus.Ok)
+        {
+            var result = magic.Execute(input, new MockChannel()).GetAwaiter().GetResult();
+            Assert.IsTrue(result.Status == expected);
+        }
+    }
+
+    [TestClass]
+    public class AzureClientMagicTests
+    {
+        private readonly string subscriptionId = "TEST_SUBSCRIPTION_ID";
+        private readonly string resourceGroupName = "TEST_RESOURCE_GROUP_NAME";
+        private readonly string workspaceName = "TEST_WORKSPACE_NAME";
+        private readonly string storageAccountConnectionString = "TEST_CONNECTION_STRING";
+        private readonly string jobId = "TEST_JOB_ID";
+        private readonly string operationName = "TEST_OPERATION_NAME";
+        private readonly string targetName = "TEST_TARGET_NAME";
+
+        [TestMethod]
+        public void TestConnectMagic()
+        {
+            var azureClient = new MockAzureClient();
+            var connectMagic = new ConnectMagic(azureClient);
+
+            // unrecognized input
+            connectMagic.Test($"invalid");
+            Assert.AreEqual(azureClient.LastAction, AzureClientAction.PrintConnectionStatus);
+
+            // valid input
+            connectMagic.Test(
+                @$"subscriptionId={subscriptionId}
+                   resourceGroupName={resourceGroupName}
+                   workspaceName={workspaceName}
+                   storageAccountConnectionString={storageAccountConnectionString}");
+            Assert.AreEqual(azureClient.LastAction, AzureClientAction.Connect);
+            Assert.IsFalse(azureClient.ForceLogin);
+            Assert.AreEqual(azureClient.ConnectionString, storageAccountConnectionString);
+
+            // valid input with forced login
+            connectMagic.Test(
+                @$"login
```

Comment:
This line is getting highlighted weird by GitHub; there may be an invisible special character here?

Reply:
VS Code is doing this too. Apparently it thinks it's an unclosed string literal (same with the one a few lines down). If I put `subscriptionId` on the same line then it goes away. I'll do that just to avoid weirdness.